### ISSUE TITLE
Improve registration modal window

### ISSUE THREAD

**Issue by mike-podolskiy90** [gbif/ipt#2590]
In Test mode message can be confusing: 

![Image](https://github.com/user-attachments/assets/da0052e6-6105-49be-95c9-cf681a91b061)

There is no point in displaying the first part (in italics) since registration is disabled due to the resource being private. So it is better to hide it.
It would also be convenient to add a button/link to make the resource public in the same modal window and explain that it should be published too.

**Comment by ManonGros** [comment]
Yes I agree, that it would make sense to have "Resource must be public in order to be registered." and display the other message only when the dataset can actually be registered.
